sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -117,6 +118,7 @@ class RocksDB(
   dbOptions.setTableFormatConfig(tableFormatConfig)
   dbOptions.setMaxOpenFiles(conf.maxOpenFiles)
   dbOptions.setAllowFAllocate(conf.allowFAllocate)
+  dbOptions.setMergeOperator(new StringAppendOperator())
```

Comment:
Its safe to set this for col families that don't need `merge` at all right ?

Reply:
yes, its safe. One aspect is that values will be encoded with a extra int64 value containing the len of actual UnsafeRow bytes. We can avoid it in future by refactoring RocksDBStateEncoder into separate encoders for Key and Value respectively.

Reply:
Oh nice - I already made this change then :)